src/librustc_mir_build/hair/pattern/_match.rs

```diff
@@ -893,11 +893,12 @@ impl<'tcx> Constructor<'tcx> {
                                     // Treat all uninhabited types in non-exhaustive variants as
                                     // `TyErr`.
                                     (_, true, true) => cx.tcx.types.err,
-                                    // Treat all non-visible fields as `TyErr`. They can't appear
-                                    // in any other pattern from this match (because they are
+                                    // Treat all non-visible fields as `()`. They can't appear in
```

Comment:
This seems arbitrary; cc @varkor @Nadrieril

Reply:
On the other hand it says that we are picking some inhabited dummy, so `()` is as good as any I suppose.